EKF/control.cpp

```diff
@@ -720,9 +730,10 @@ void Ekf::controlHeightFusion()
 				}
 			}
 
-		} else if (_baro_data_ready && !_baro_hgt_faulty && !_in_range_aid_mode) {
+		} else if (_baro_data_ready && !_baro_hgt_faulty && _rng_hgt_faulty) {
```

Comment:
should be
```
else if (_baro_data_ready && !_baro_hgt_faulty &&
!(_in_range_aid_mode && !_range_data_ready && !_rng_hgt_faulty))
```
to prevent switching to baro if in range aid, rng is not faulty but not ready